ecl/resource_ecl_provider_connectivity_tenant_connection_v2.go

```diff
@@ -0,0 +1,461 @@
+
+	"github.com/nttcom/eclcloud"
+	"github.com/nttcom/eclcloud/ecl/provider_connectivity/v2/tenant_connections"
+)
+
+func resourceProviderConnectivityTenantConnectionV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceProviderConnectivityTenantConnectionV2Create,
+		Read:   resourceProviderConnectivityTenantConnectionV2Read,
+		Update: resourceProviderConnectivityTenantConnectionV2Update,
+		Delete: resourceProviderConnectivityTenantConnectionV2Delete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"tenant_connection_request_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"device_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"ECL::Compute::Server",
+					"ECL::Baremetal::Server",
+					"ECL::VirtualNetworkAppliance::VSRX",
+				}, false),
+			},
+			"device_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"device_interface_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"attachment_opts_server": &schema.Schema{
+				Type:          schema.TypeList,
+				Optional:      true,
+				MaxItems:      1,
+				ConflictsWith: []string{"attachment_opts_vna"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"segmentation_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"segmentation_id": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"fixed_ips": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ip_address": &schema.Schema{
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.SingleIP(),
+									},
+									"subnet_id": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"allowed_address_pairs": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ip_address": &schema.Schema{
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.SingleIP(),
+									},
+									"mac_address": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"attachment_opts_vna": &schema.Schema{
+				Type:          schema.TypeList,
+				Optional:      true,
```

Comment:
```suggestion
				Optional:      true,
				ForceNew: true,
```

Reply:
Related to this.
https://github.com/nttcom/terraform-provider-ecl/pull/46#discussion_r375687948